core/domain/expression_parser.py

```diff
@@ -0,0 +1,541 @@
+            parsed_right = self.parse_pow_expr()
+            return PowerOperator(parsed_expr, parsed_right)
+        return parsed_expr
+
+    def parse_unit(self):
+        """Function representing the following production rule of the grammar:
+        <unit> ::= <identifier> | <number> | '(' <expr> ')' |
+        <function> '(' <expr> ')'
+
+        Returns:
+            Node. Root node of the generated parse tree.
+
+        Raises:
+            Exception: Invalid token.
+        """
+
+        token = self.get_next_token()
+        if token.category == 'identifier':
+            return Identifier(token)
+
+        if token.category == 'function':
+            if self.is_next_token(['(']):
+                parsed_child = self._parse_expr()
+                self.expect_token(')')
+                return Function(token, parsed_child)
+
+        if token.category == 'number':
+            return Number(token)
+
+        if token.text == '(':
+            parsed_expr = self._parse_expr()
+            self.expect_token(')')
+            return parsed_expr
+
+        raise Exception('Invalid token: %s.' % token.text)
+
+    def check_next_token(self):
+        """Function to peek into the next position to see the next token.
+
+        Returns:
+            Token|None. Token at the next position. Returns none if there are no
+                more tokens left.
+        """
+
+        if self.next_token_index < len(self.token_list):
+            return self.token_list[self.next_token_index]
+
+    def get_next_token(self):
+        """Function to retrieve the token at the next position and then
+        increment the next_token_index.
+
+        Returns:
+            Token. Token at the next position.
+
+        Raises:
+            Exception: Invalid syntax: Unexpected end of expression.
+        """
+
+        if self.next_token_index < len(self.token_list):
+            token = self.token_list[self.next_token_index]
+            self.next_token_index += 1
+            return token
+
+        raise Exception('Invalid syntax: Unexpected end of expression.')
+
+    def is_next_token(self, allowed_token_texts):
+        """Function to verify that there is at least one more token remaining
+        and that the next token is among the allowed_token_texts provided.
+
+        Args:
+            allowed_token_texts: list(str). List of strings containing the
+                allowed token texts at the next position.
+
+        Returns:
+            Token|None. Token at the next position. Returns none if there are no
+                more tokens left or the next token is not in the
+                allowed_token_texts.
+        """
+
+        if self.next_token_index < len(self.token_list):
+            text = self.token_list[self.next_token_index].text
+            if text in allowed_token_texts:
+                token = self.token_list[self.next_token_index]
+                self.next_token_index += 1
+                return token
```

Comment:
Just return True? Since the function is just used for checking.

Reply:
I've renamed the function to avoid confusion.